internal-enrichment/greynoise-vuln/src/main.py

```diff
@@ -0,0 +1,506 @@
+
+        if "exploitation_activity" in data:
+            self._generate_stix_note(stix_entity, data)
+        self._generate_stix_software(stix_entity, data)
+        self._generate_stix_vulnerability(data, labels, external_reference)
+
+        uniq_bundles_objects = list(
+            {obj["id"]: obj for obj in self.stix_objects}.values()
+        )
+
+        self.helper.connector_logger.info(
+            "[CONNECTOR] For this CVE, the number of Stix bundle(s) that will be enriched.",
+            {
+                "Vulnerability": stix_entity["name"],
+                "Stix_bundle_length": len(uniq_bundles_objects),
+            },
+        )
+
+        stix2_bundle = self.helper.stix2_create_bundle(uniq_bundles_objects)
+        return stix2_bundle
+
+    def _process_message(self, data: Dict) -> str:
+        # Security to limit playbook triggers to something other than the scope initial
+        scopes = self.helper.connect_scope.lower().replace(" ", "").split(",")
+        entity_splited = data["entity_id"].split("--")
+        entity_type = entity_splited[0].lower()
+
+        if not self.check_api_key():
+            self.helper.log_error(
+                "GreyNoise API Key is NOT valid. Update to Enterprise API key to use this connector."
+            )
+            raise ValueError(
+                "GreyNoise API Key is NOT valid. Update to Enterprise API key to use this connector."
+            )
+
+        if entity_type in scopes:
+            # OpenCTI entity information retrieval
+            stix_entity = data["stix_entity"]
+            opencti_entity = data["enrichment_entity"]
+            self.stix_objects = data["stix_objects"]
+
+            is_valid_max_tlp = self._extract_and_check_markings(opencti_entity)
+            if not is_valid_max_tlp:
+                raise ValueError(
+                    "[ERROR] Do not send any data, TLP of the observable is greater than MAX TLP, "
+                    "the connector does not has access to this observable, please check the group of the connector user"
+                )
+
+            # Extract Value from opencti entity data
+            opencti_entity_value = stix_entity["name"]
+
+            try:
+                # Get "CVE Context" GreyNoise API Response
+                # https://docs.greynoise.io/reference/get_v1-cve-cve-id
+                self.helper.connector_logger.info(
+                    f"Get CVE context for: {opencti_entity_value}"
+                )
+                session = GreyNoise(
+                    api_key=self.greynoise_key, integration_name=self.integration_name
+                )
+
+                json_data = session.cve(opencti_entity_value)
+
+                # Generate a stix bundle
+                stix_bundle = self._generate_stix_bundle(json_data, stix_entity)
+
+                # Send stix2 bundle
+                bundles_sent = self.helper.send_stix2_bundle(stix_bundle)
```

Comment:
When a CVE is not found, the response "CVE not found" is raised. 

![image](https://github.com/user-attachments/assets/5e6be824-7a70-46d2-9944-3ea26cd0e566)

This case needs to be handled too.

Example of simple condition:

```suggestion
                json_data = session.cve(opencti_entity_value)

                # Generate a stix bundle
                stix_bundle = None
                
                if "CVE not found" == json_data:
                    stix_bundle = self.helper.stix2_create_bundle(self.stix_objects)
                    bundles_sent = self.helper.send_stix2_bundle(stix_bundle)
                    return "[CONNECTOR] No CVE found. Original Vulnerability sent:  "
                      + str(len(bundles_sent))
                      + " stix bundle(s) for worker import"
                   )
                else:
                   stix_bundle = self._generate_stix_bundle(json_data, stix_entity)
                   bundles_sent = self.helper.send_stix2_bundle(stix_bundle)
                   return (
                      "[CONNECTOR] Sent "
                      + str(len(bundles_sent))
                      + " stix bundle(s) for worker import"
                   )


```